src/GitHub/Data/Actions.hs

```diff
@@ -0,0 +1,97 @@
+-----------------------------------------------------------------------------
+-- |
+-- License     :  BSD-3-Clause
+-- Maintainer  :  Oleg Grenrus <oleg.grenrus@iki.fi>
+--
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE KindSignatures    #-}
+module GitHub.Data.Actions (
+    Artifact(..),
+    ArtifactList(..),
+    Workflow,
+    PaginatedWithTotalCount(..),
+    WithTotalCount(..),
+    WorkflowRun,
+    ) where
+
+import GHC.TypeLits
+import GitHub.Data.Id          (Id)
+import GitHub.Data.URL         (URL)
+import GitHub.Internal.Prelude
+import Prelude ()
+
+import           Data.Data (Proxy (..))
+import qualified Data.Text as T
+
+data Artifact = Artifact
+    { artifactArchiveDownloadUrl :: !URL
+    , artifactCreatedAt :: !UTCTime
+    , artifactExpired :: !Bool
+    , artifactExpiresAt :: !UTCTime
+    , artifactId :: !(Id Artifact)
+    , artifactName :: !Text
+    , artifactNodeId :: !Text
+    , artifactSizeInBytes :: !Int
+    , artifactUpdatedAt :: !UTCTime
+    , artifactUrl :: !URL
+    }
+  deriving (Show, Data, Typeable, Eq, Ord, Generic)
+
+data ArtifactList = ArtifactList
+    { artifactListArtifacts :: !(Vector Artifact)
+    , artifactListTotalCount :: !Int
+    }
+    deriving (Show, Data, Typeable, Eq, Ord, Generic)
+
+data PaginatedWithTotalCount a (tag :: Symbol) = PaginatedWithTotalCount
```

Comment:
What are these? Is artifacts pagination different then for everything else?

Reply:
Yes, please check my comment above. At least I didn't find existing implementation for this pagination style.